direnv.el

```diff
@@ -230,6 +262,25 @@ Same as ‘provided-mode-derived-p’ which is Emacs 26.1+ only."
 (when (fboundp 'provided-mode-derived-p)
   (defalias 'direnv--provided-mode-derived-p 'provided-mode-derived-p))
 
+(defun direnv--callback (items show-summary old-directory)
+  (let ((summary (direnv--summarise-changes items)))
+    (when (and direnv-always-show-summary (not (string-empty-p summary)))
+      (setq show-summary t))
+    (when show-summary
+      (direnv--show-summary summary old-directory direnv--active-directory))
```

Comment:
The timing of this has changed substantially. Does this need to be passed as an argument to guard against the value changing? Or do we not really care?